### DISCUSSION TITLE
How do you revert someone's commit?

### DISCUSSION THREAD

**Discussion by AppDevUW** [UWAppDev/community#4] [Q&A] [answered]
Question from Discord server

**Comment by ApolloZhu** [comment, marked as answer]
There are several ways you could do this:

## 1. Revert but keep it in history

Let's say you'd like to revert commit `9fb4114d` "fix casing" (even if you didn't author it), you can do:

```sh
git revert 9fb4114d
```

... and all the changes introduced in commit `9fb4114d` will be reverted:

![image](https://user-images.githubusercontent.com/10842684/127723577-4972902b-120e-4757-bc92-15e01c8ea771.png)

If you use git GUI, most applications provide an option to revert a commit (likely in the right click menu).

---

> ```diff
> ! IMPORTANT NOTICE
> ```
> For the following 2 options, you might need to [force push](https://www.git-tower.com/learn/git/faq/git-force-push/) to the server if the commits you are changing is already on the server. Make sure others are aware of the force push so they can make necessary adjustments/updates.

## 2. Rewrite history

*Interactive rebase* is more advanced, but you could achieve many things with it: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

## 3. Undo recent commits

- Amend most recent commit: see article linked in "2. Rewrite history".
- (Soft) reset recent commits: https://stackoverflow.com/questions/927358/how-do-i-undo-the-most-recent-local-commits-in-git
